debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/view/dtfj/java/DTFJJavaRuntime.java

```diff
@@ -929,13 +929,21 @@ public String getFullVersion() throws CorruptDataException {
 	}
 
 	public String getVersion() throws CorruptDataException {
+		String version = null;
 		try {
 			//starting with 26 stream, RAS structure contains the JRE version which avoids the potential pitfall
 			//of using system properties which may have their name and/or value stored in native libraries
-			return DTFJContext.getVm().j9ras().serviceLevel().getCStringAtOffset(0);
+			version = DTFJContext.getVm().j9ras().serviceLevel().getCStringAtOffset(0);
 		} catch (Throwable t) {
 			throw J9DDRDTFJUtils.handleAsCorruptDataException(DTFJContext.getProcess(), t);
 		}
+		try {
+			String productName = DTFJContext.getVm().j9ras().productName().getCStringAtOffset(0);
+			version += " " + productName;
+		} catch (Throwable t) {
+			// ignore if the product name doesn't exist
+		}
```

Comment:
Suggest capturing `j9ras()` and reusing it (within the existing `try` block):
```java
			J9RASPointer ras = DTFJContext.getVm().j9ras();
			version = ras.serviceLevel().getCStringAtOffset(0);
			try {
				U8Pointer productName = ras.productName();
				if (productName.notNull()) {
					version = version + " " + productName.getCStringAtOffset(0);
				}
			} catch (NoSuchFieldException e) {
				// ignore if the product name doesn't exist
			}
```
Note the narrower catch exception type, the explicit null check, and the (slightly) less expensive concatenation.